sdk/storage/azure_storage_queue/src/clients/queue_service_client.rs

```diff
@@ -0,0 +1,204 @@
+    ///
+    /// Returns a `QueueClient` that can be used to interact with the specified queue.
+    pub fn queue_client(&self, queue_name: String) -> QueueClient {
+        QueueClient {
+            endpoint: self.endpoint.clone(),
+            client: self.client.get_queue_client(queue_name),
+        }
+    }
+
+    /// Creates a new queue under the given account.
+    ///
+    /// # Arguments
+    ///
+    /// * `options` - Optional parameters for the request
+    /// * `queue_name` - The name of the queue to create
+    ///
+    /// # Returns
+    ///
+    /// Returns a `Result` containing the response if successful
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the queue already exists or if the request fails
+    pub async fn create_queue(
+        &self,
+        queue_name: &str,
+        options: Option<QueueClientCreateOptions<'_>>,
+    ) -> Result<Response<(), NoFormat>> {
+        self.client
+            .get_queue_client(queue_name.to_string())
+            .create(options)
+            .await
+    }
+
+    /// Permanently deletes the specified queue.
+    ///
+    /// # Arguments
+    ///
+    /// * `options` - Optional parameters for the request
+    /// * `queue_name` - The name of the queue to create
+    ///
+    /// # Returns
+    ///
+    /// Returns a `Result` containing the response if successful
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the queue doesn't exist or if the request fails
+    pub async fn delete_queue(
+        &self,
+        queue_name: &str,
+        options: Option<QueueClientDeleteOptions<'_>>,
+    ) -> Result<Response<(), NoFormat>> {
+        self.client
+            .get_queue_client(queue_name.to_string())
+            .delete(options)
+            .await
+    }
+
+    /// Retrieves the properties of the queue service.
+    ///
+    /// # Arguments
+    ///
+    /// * `options` - Optional parameters for the request
+    ///
+    /// # Returns
+    ///
+    /// Returns a `Result` containing the service properties response if successful
+    ///
+    /// # Note
+    ///
+    /// This returns properties for the entire service, not just a single queue.
+    pub async fn get_properties(
+        &self,
+        options: Option<QueueServiceClientGetPropertiesOptions<'_>>,
+    ) -> Result<Response<StorageServiceProperties, XmlFormat>> {
+        self.client.get_properties(options).await
+    }
+
+    /// Sets the properties of the queue service.
+    ///
+    /// # Arguments
+    ///
+    /// * `storage_service_properties` - The properties to set for the queue service
+    /// * `content_type` - The content type of the request body, typically "application/xml"
+    /// * `options` - Optional parameters for the request
+    ///
+    /// # Returns
+    ///
+    /// Returns a `Result` containing the response if successful.
+    pub async fn set_properties(
+        &self,
+        storage_service_properties: RequestContent<StorageServiceProperties>,
```

Comment:
Can we rename this model to just `QueueServiceProperties`? This would also change `get_properties`. This can be done in the TypeSpec and we will do the same for Blob. Don't make the TypeSpec change specific to Rust, fine to completely rename the model.